django/db/models/fields/related.py

```diff
@@ -1885,6 +1891,13 @@ def __init__(self, to, to_field=None, related_name=None, related_query_name=None
             # be correct until contribute_to_class is called. Refs #12190.
             to_field = to_field or (to._meta.pk and to._meta.pk.name)
 
+        if on_delete is None:
```

Comment:
`ForeignKey` inherits `ForeignObject` and calls the super `__init__` below, so if we have the warning in `ForeignObject` why do we also need it here in `ForeignKey`?